src/web-components.js

```diff
@@ -48,6 +48,26 @@ export function isShadowDomSupported() {
 }
 
 /**
+ * Returns `true` if Shadow CSS encapsulation is supported.
+ * @return {boolean}
+ */
+export function isShadowCssSupported() {
+  if (!isShadowDomSupported()) {
+    return false;
+  }
+
+  //TODO: Find a better way to test CSS encapsulation
+  if (window.ShadyCSS) {
```

Comment:
We need to fully anticipate that we won't be able to flag this value "statically". Though it doesn't mean we can't cache it statically here. But it does mean that we may pass `ShadowRoot` instance in this method to check if it supports shadow CSS. Some ideas to test this:

1. Try to reach host element from `shadowRoot` via any DOM APIs possible (except `.host` of course).
2. Super heavy and might force us to be async: insert empty `<style>` inside the `shadowRoot` and see if it also ended up in the `document.styleSheets`.
3. Might be simplest, but need to check: `Object.prototype.toString.call(shadowRoot)` will likely return `[object DocumentFragment]` instead of `[object ShadowRoot]` for polyfills.
4. etc